src/test/java/com/example/project/test/ShoppingListTest.java

```diff
@@ -0,0 +1,138 @@
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public class ShoppingListTest {
+
+    AndroidDriver<AndroidElement> driver;
+
+    @BeforeClass
+    public void setUp() throws MalformedURLException {
+        DesiredCapabilities cap = new DesiredCapabilities();
+        cap.setCapability("deviceName", "sdk_gphone_x86_arm");
+        cap.setCapability("udid", "emulator-5554");
+        cap.setCapability("platformName", "Android");
+        cap.setCapability("platformVersion", "11");
+
+        cap.setCapability("appPackage", "com.slava.buylist");
+        cap.setCapability("appActivity", "com.slava.buylist.MainActivity");
+
+        URL url = new URL("http://127.0.0.1:4723/wd/hub");
+        driver = new AndroidDriver<>(url, cap);
+
+        driver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);
+    }
+
+    @BeforeMethod
+    public void resetApp() {
+        driver.resetApp();
+        driver.findElementById("com.android.permissioncontroller:id/continue_button").click();
+        driver.findElementById("android:id/button1").click();
+    }
+
+    @Test
+    public void createNewList_successful() throws InterruptedException {
+        final String listName = "New list";
+        AllListsPage allListsPage = createNewList(listName);
+        Thread.sleep(2000);
+        driver.hideKeyboard();
+        driver.navigate().back();
+        assertThat(allListsPage.firstListNameTextView.getText()).isEqualTo(listName);
+    }
+
+    @Test
+    public void addItemToList_successful() {
+        createNewList("New list");
+
+        waitForListPageToLoad();
+
+        ListPage listPage = addItemToList("Milk", "1.2", "2", "4", "2");
+
+        assertThat(listPage.firstItemNameTextView.getText()).isEqualTo("Milk");
+    }
+
+    @Test
+    public void editItem() {
+        createNewList("New list");
+
+        waitForListPageToLoad();
+
+        ListPage listPage = addItemToList("Milk", "1.2", "2", "4", "2")
+                .longPressOnFirstItem()
+                .clickOnEdit()
+                .enterComment("3% fat")
+                .enterPrice("1.3")
+                .saveItem();
+
+        assertThat(listPage.firstItemPriceTextView.getText()).isEqualTo("1.3 £");
+        assertThat(listPage.firstItemCommentTextView.getText()).isEqualTo("3% fat");
+    }
+
+    @Test
+    public void removeItem() {
+        createNewList("New list");
+
+        waitForListPageToLoad();
+
+        addItemToList("Milk", "1.2", "2", "4", "2")
```

Comment:
would it be possible to use chaining for the above steps as well? so that we stick to the same style through the test